src/controller/java/BaseCHIPCluster-JNI.cpp

```diff
@@ -3,57 +3,659 @@
 #include <lib/support/JniReferences.h>
 #include <platform/PlatformManager.h>
 
+#include "AndroidCallbacks.h"
+
 #define JNI_METHOD(RETURN, CLASS_NAME, METHOD_NAME)                                                                                \
     extern "C" JNIEXPORT RETURN JNICALL Java_chip_devicecontroller_ChipClusters_00024##CLASS_NAME##_##METHOD_NAME
 
-JNI_METHOD(void, BaseChipCluster, deleteCluster)(JNIEnv * env, jobject self, jlong clusterPtr)
+#define CHIP_TLV_WRITER_BUFFER_SIZE 1024
+
+JNI_METHOD(void, BaseChipCluster, readAttribute)
```

Comment:
can we remove [BaseCHIPCluster-JNI.cpp](https://github.com/project-chip/connectedhomeip/pull/30141/files#diff-b20dc1000c6332c76f0eff8502342ac71eaff1acc32971cc890c1ab53c838c98)?

is it possible that the generated java API use generic java im api directly?





[9:09](https://csamembers.slack.com/archives/D03QTDMAA5R/p1698811777603699)
is it possible that the generated java API use generic java im api directly?

Reply:
What is the difference between generic JNI vs generic IM API, because of lacking of TLV support in Java version?

Reply:
the existing API support tlv/cluster object( kind of native type)/json...so technically it should be fine, the generated java api needs native type instead of tlv

Reply:
BaseCHIPCluster-JNI.cpp cannot be deleted due to TLV encoding and decoding. However, I will change the other methods to use those in the existing CHIPDeviceController class.

Reply:
Altering the existing CHIPDeviceController class to utilize the refactored interfaces will not address the issue. Instead, it make CHIPDeviceController dependent on BaseCHIPCluster-JNI.cpp. My preference is to use BaseCHIPCluster-JNI.cpp solely for Cluster API interactions while leaving CHIPDeviceController as it is, without modifications.

I agree it is hard to use generic Java IM API directly since no reference to CHIPDeviceController.

Reply:
Yes, I agree with that. I'll think about it.